src/ORM/EagerLoadedDataList.php

```diff
@@ -0,0 +1,893 @@
+<?php
+
+namespace SilverStripe\ORM;
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\View\ViewableData;
+use SilverStripe\ORM\Connect\DatabaseException;
+use SilverStripe\ORM\FieldType\DBField;
+use BadMethodCallException;
+use InvalidArgumentException;
+use LogicException;
+use Traversable;
+
+/**
+ * Represents an "eager loaded" DataList - i.e. the data has already been fetched from the database
+ * for these records and likely for some of their relations.
+ *
+ * This list is designed to be plug-and-play with the various DataList implementations, with the exception
+ * that because it doesn't make a database query to get its data, some methods are intentionally not implemented.
+ *
+ * Note that when this list represents a relation, adding items to or removing items from this list will NOT
+ * affect the underlying relation in the database. This list is read-only.
+ */
+class EagerLoadedDataList extends ViewableData implements Relation, SS_List, Filterable, Sortable, Limitable
```

Comment:
```suggestion
class EagerLoadedRelationList extends ViewableData implements Relation, SS_List, Filterable, Sortable, Limitable
```
Would this renaming the class to `EagerLoadedRelationList` make sense? That would do away with the need to make it mirror DataList.

Looking at this now it's confusing to me having the word 'DataList' in here since it's not a DataList, and infact it has a `private DataList $dataList` which its super weird - why would a DataList need a private DataList?

We could also take it further to just `EagerLoadedList` - this was briefly discussed early though rejected because it's less nice to say out loud :-)  However if we remove 'Relation' then we could probably just remove the ' implements Relation' bit and associated 3x methods? Perhaps that would be better?

Reply:
As discussed offline, `EagerLoadedList` is preferred.

Reply:
Done